packages/dapp-tic-tac-toe/src/Waiting.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import Node from "./Node";
+
+/**
+ * User Story
+ * Bob(Proposing) waits for Alice(Accepting) to install the game
+ */
+export default class Waiting extends Component {
+  /**
+   * Bob(Proposing) enters waiting room.
+   * Bob(Proposing) makes a call to Playground for matchmaking and waits to get an Accepting player.
+   * Bob(Proposing) makes a call to CF.js proposeInstall.
+   * Bob(Proposing) waits for Alice(Accepting) to approve -- Add Waiting Room (Waiting for Alice) --
+   */
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+      seconds: 5
+    };
+  }
+
+  componentDidMount() {
+    this.countDown();
+    this.initialize();
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.countdownTimeout);
+  }
+
+  async initialize() {
+    await this.props.nodeProvider.connect();
```

Comment:
any reason you don't do this up in the `App` class? there's no need for anything but the `cfProvider` to have a reference to it

Reply:
It was to prevent the Waiting component from attempting to send a message before a connection was established. I went ahead and moved `connect` to App, which I agree makes more sense architecturally. Since the child components may still attempt to send messages before a connection is established, though, I compromised by locking the screen until after `connect` resolves.

Reply:
Removed the types, though unfortunately we've got to keep the package as a dep because other packages depend on it.